x-pack/legacy/plugins/apm/public/utils/formatters/duration.ts

```diff
@@ -0,0 +1,166 @@
+  };
+}
+
+const durationUnit: DurationTimeUnit = {
+  hours: {
+    label: i18n.translate('xpack.apm.formatters.hoursTimeUnitLabel', {
+      defaultMessage: 'h'
+    }),
+    convert: (value: number) =>
+      asDecimal(moment.duration(value / 1000).asHours())
+  },
+  minutes: {
+    label: i18n.translate('xpack.apm.formatters.minutesTimeUnitLabel', {
+      defaultMessage: 'min'
+    }),
+    convert: (value: number) =>
+      asDecimal(moment.duration(value / 1000).asMinutes())
+  },
+  seconds: {
+    label: i18n.translate('xpack.apm.formatters.secondsTimeUnitLabel', {
+      defaultMessage: 's'
+    }),
+    convert: (value: number) =>
+      asDecimal(moment.duration(value / 1000).asSeconds())
+  },
+  milliseconds: {
+    label: i18n.translate('xpack.apm.formatters.millisTimeUnitLabel', {
+      defaultMessage: 'ms'
+    }),
+    convert: (value: number) =>
+      asInteger(moment.duration(value / 1000).asMilliseconds())
+  },
+  microseconds: {
+    label: i18n.translate('xpack.apm.formatters.microsTimeUnitLabel', {
+      defaultMessage: 'μs'
+    }),
+    convert: (value: number) => asInteger(value)
+  }
+};
+
+function convertTo({
+  unit,
+  value,
+  withUnit,
+  defaultValue
+}: {
+  unit: TimeUnit | 'microseconds';
+  value: Maybe<number>;
+  withUnit: boolean;
+  defaultValue: string;
+}) {
+  const duration = durationUnit[unit];
+  if (!duration || value == null) {
+    return defaultValue;
+  }
+
+  const message = SPACE + duration.label;
+
+  const convertedValue = duration.convert(value);
+  return `${convertedValue}${withUnit ? message : ''}`;
+}
+
+export const asHours = (
+  value: Maybe<number>,
+  { withUnit = true, defaultValue = NOT_AVAILABLE_LABEL }: FormatterOptions = {}
+) => convertTo({ unit: 'hours', value, withUnit, defaultValue });
+
+export const asMinutes = (
+  value: Maybe<number>,
+  { withUnit = true, defaultValue = NOT_AVAILABLE_LABEL }: FormatterOptions = {}
+) => convertTo({ unit: 'minutes', value, withUnit, defaultValue });
+
+export const asSeconds = (
+  value: Maybe<number>,
+  { withUnit = true, defaultValue = NOT_AVAILABLE_LABEL }: FormatterOptions = {}
+) => convertTo({ unit: 'seconds', value, withUnit, defaultValue });
+
+export const asMillis = (
+  value: Maybe<number>,
+  { withUnit = true, defaultValue = NOT_AVAILABLE_LABEL }: FormatterOptions = {}
+) => convertTo({ unit: 'milliseconds', value, withUnit, defaultValue });
+
+export const asMicros = (
+  value: Maybe<number>,
+  { withUnit = true, defaultValue = NOT_AVAILABLE_LABEL }: FormatterOptions = {}
+) => convertTo({ unit: 'microseconds', value, withUnit, defaultValue });
+
+export type TimeFormatter = (
+  value: Maybe<number>,
+  options?: FormatterOptions
+) => string;
+
+type TimeFormatterBuilder = (max: number) => TimeFormatter;
+
+export const getDurationFormatter: TimeFormatterBuilder = memoize(
+  (max: number) => {
+    const unit = getDurationUnit(max);
+    switch (unit) {
+      case 'h':
```

Comment:
We have two different unit identifiers: "h" and "hours". Would it make sense to combine them.
(I haven't looked into this, just noticed and thought it looked a bit inconsistent)